include/alpaka/dev/DevCpu.hpp

```diff
@@ -176,6 +183,17 @@ namespace alpaka
                 // The CPU does nothing on reset.
             }
         };
+
+        //! The CPU device native handle type trait specialization.
+        template<>
+        struct NativeHandle<DevCpu>
+        {
+            using type = int;
```

Comment:
I don't see where this `type` is used. Can we remove it?

Reply:
@tonydp03 Does CMS need this?

Reply:
Well, for now the function just returns 0, so we don't really needed. But in a previous discussion, conveyed that it's useful to have the same interface for all the devices (also we were discussing about handling multi-CPU machines in a different way in future). Could it be useful to keep it for now or we'll modify it at the right time?

Reply:
I would remove it for now since the user usually doesn't query the trait directly but uses the free function instead. If we decide later that this is required it is not too hard to put in.

Reply:
> @tonydp03 Does CMS need this?

Yes, to use the common interface to get the "native handle" of a device, we also need a common way to get its type.

Reply:
Alright, I don't really have a strong opinion on this. @bernhardmgruber, what do you think?

Reply:
I strongly disagree with the suggestion that `decltype(alpaka::getNativeHandle(std::declval<Dev>()))` is as user friendly as `alpaka::NativeHandle<Dev>` .

Reply:
I did not claim my suggestion is user friendly, but I explained why this is not a primary concern.

Then, `typename alpaka::traits::NativeHandle<Dev>::type` would give you the type. But now the user has to grab into the `alpaka::traits` which is our namespace for customization points. This is not what we want the users to do.

If you would really like to have that alias, then how about adding:
```c++
namespace alpaka {
  template <typename Dev>
  using NativeHandle = decltype(getNativeHandle(std::declval<Dev>()));
}
```
and still removing the nested `type` alias in the `alpaka::traits::NativeHandle` specializations?

Reply:
I do not mind how the `alpaka::NativaHandle<Dev>` type is made available.

From what I can see, everything else in alpaka is exposed with a nested template type or function in `alpaka::traits`, and an alias in the top level `alpaka` namespace. If you think this one case is better implemented differently, sure, why not.